src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt

```diff
@@ -386,6 +386,28 @@ class AssessmentService(
     )
   }
 
+  private val cruMappings = mutableMapOf<String, List<String>>(
+    "Midlands" to listOf("N31", "N53"),
+    "South East & Eastern incl. Women" to listOf("N34", "N56"),
+    "North West" to listOf("N28", "N50", "MCG", "GMP"),
+    "South East & Eastern incl. Women" to listOf("N35", "N57"),
+    "London" to listOf("N07", "N21", "C17", "LDN"),
+    "North East" to listOf("N32", "N54", "N02", "N23"),
+    "North West" to listOf("N29", "N51", "N24", "N01"),
+    "Wales" to listOf("N03", "C10", "N27", "WPT"),
+    "Midlands" to listOf("MLW", "N30", "N52"),
+    "North West" to listOf("N33", "N55"),
+    "South West & South Central" to listOf("N36", "N58", "N26", "N37", "N59", "N05")
+  )
+
+  private fun cruNameFromProbationAreaCode(code: String): String {
+    cruMappings.forEach {
+      if (it.value.contains(code)) return it.key
+    }
+
+    throw RuntimeException("Could not find CRU Name for Probation Area Code: $code")
+  }
```

Comment:
I think we probably want to have a unit test to describe the behaviour we want if the assessor has a probation area which doesn't map to one of our CRUs?

At present I think we'll fail hard? No assessment submission, no event etc, right?

As this tracking of the CRU responsible for assessment is used for performance monitoring rather than by Delius it feels more appropriate to catch the error and for example:

- have a default CRU value of something like `CRU not identified`
- log an error like `CRU matching probation area name: 'Staffordshire and West Midlands' code: 'SWM' not found`

Performance monitoring should be able to handle these anomalies by doing their own mapping using the probation area which is reported.

Reply:
Makes sense, have changed to return "Unknown CRU" when the mapping is missing + added a test for this